spring-cloud-bindings/src/main/java/org/springframework/cloud/bindings/boot/ConfigServerBindingsPropertiesProcessor.java

```diff
@@ -40,12 +50,63 @@ public void process(Environment environment, Bindings bindings, Map<String, Obje
         }
 
         bindings.filterBindings(TYPE).forEach(binding -> {
-            MapMapper map = new MapMapper(binding.getSecret(), properties);
+            Map<String, String> secret = binding.getSecret();
+            MapMapper map = new MapMapper(secret, properties);
             map.from("uri").to("spring.cloud.config.uri");
             map.from("client-id").to("spring.cloud.config.client.oauth2.clientId");
             map.from("client-secret").to("spring.cloud.config.client.oauth2.clientSecret");
             map.from("access-token-uri").to("spring.cloud.config.client.oauth2.accessTokenUri");
+
+            // When tls.crt and tls.key are set, enable mTLS for config client.
+            String clientKey = secret.get("tls.key");
+            String clientCert = secret.get("tls.crt");
+            if (StringUtils.hasText(clientCert) != StringUtils.hasText(clientKey)) {
+                throw new IllegalArgumentException("binding secret error: tls.key and tls.crt must both be set if either is set");
+            }
+
+            if (clientKey != null && !clientKey.isEmpty()) {
+                String generatedPassword = new Random().ints(97 /* letter a */, 122 /* letter z */ + 1)
```

Comment:
could you rather extract this as a method in `PemSslStoreHelper` so that we don't duplicate those 4 lines across `EurekaBindingsPropertiesProcessor` and `ConfigServerBindingsPropertiesProcessor` ?
Thanks

Reply:
done, together with some other methods which are moved to PemSslStoreHelper.

Reply:
thank you! looks great!